src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -201,16 +201,10 @@ TargetWorkerSetNodeList(TargetWorkerSet targetWorkerSet, LOCKMODE lockMode)
 	List *workerNodeList = ActivePrimaryNonCoordinatorNodeList(lockMode);
 	List *result = NIL;
 
-	int32 localGroupId = GetLocalGroupId();
-
 	WorkerNode *workerNode = NULL;
 	foreach_ptr(workerNode, workerNodeList)
 	{
-		if (targetWorkerSet == WORKERS_WITH_METADATA && !workerNode->hasMetadata)
-		{
-			continue;
-		}
-		if (targetWorkerSet == OTHER_WORKERS && workerNode->groupId == localGroupId)
```

Comment:
Hmm, it looks like we never use `OTHER_WORKERS `, good catch

Reply:
This was realized by @JelteF https://github.com/citusdata/citus/pull/4005#discussion_r452254384